portal_spa/src/components/forms/relying-party/RelyingPartyTabs.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from "@/components/ui/button";
+import { useState } from "react";
+import DnsChallenges from "./child-components/DNSCheck";
+import RelyingPartyForm from "./child-components/RelyingPartyForm";
+import { useRelyingParty } from "@/contexts/relying-party/RelyingPartyContext";
+
+export default function RelyingPartyTabs() {
+  const { defaultValues, isEditMode } = useRelyingParty();
+
+  const hasValidHostnames =
+    isEditMode &&
+    Array.isArray(defaultValues.hostnames) &&
+    defaultValues.hostnames.some(
+      (h: { hostname: string }) => h.hostname?.trim() !== ""
```

Comment:
According to the type, hostname won't ever be empty. This will mark it as optional.

```suggestion
      (h: { hostname?: string }) => h.hostname?.trim() !== ""
```